framework/src/meshmodifiers/SideSetsBetweenSubdomains.C

```diff
@@ -52,26 +52,32 @@ SideSetsBetweenSubdomains::modify()
 
   MeshBase::const_element_iterator   el  = mesh.active_elements_begin();
   const MeshBase::const_element_iterator end_el = mesh.active_elements_end();
-  for (; el != end_el ; ++el)
+
+  for (unsigned int i = 0; i < master_ids.size(); ++i)
```

Comment:
So what I changed now are
1. to use std::vector<SubdomainID> vec_master_ids and vec_paired_ids and convert them to std::set
2. looping over element and side and do what you suggested. But what I got confused is the last paragraph.

So to me, master_ids.count(curr_subdomain) > 0 && paired_ids.count(neighbor->subdomain_id()) > 0 checks if curr_subdomain and neighbor->subdomain_id() are in master_ids and paired_ids respectively. To me, it looks like, since we are checking every two contiguous elements by every side of the all elements, those subdomains without any contact side will be excluded, automatically, which means no empty sides are added for non-contiguous subdomains. Does this make sense?